### PR TITLE
feat(data): Add Global Configuration

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

Add Global Configuration

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/Contrib/Configuration/Masa.Contrib.Configuration/Extensions/WebApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -5,6 +5,46 @@ namespace Microsoft.AspNetCore.Builder;
...
+    {
+        if (builder.Services.Any(service => service.ImplementationType == typeof(InitializeAppConfigurationProvider)))
+            return builder;
+
+        builder.Services.AddSingleton<InitializeAppConfigurationProvider>();
+
+        MasaAppConfigureOptionsRelation optionsRelation = new();
+        action?.Invoke(optionsRelation);
+        IConfiguration configuration = builder.Configuration;
+        bool isInitialize = false;
```
Comment (doddgu) [inline src/Contrib/Configuration/Masa.Contrib.Configuration/Extensions/WebApplicationBuilderExtensions.cs:23]: isInitialize -> initialized

File: src/Contrib/Configuration/Masa.Contrib.Configuration/Options/MasaAppConfigureOptionsRelation.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.Configuration.Options;
+
+public class MasaAppConfigureOptionsRelation
+{
+    public Dictionary<string, string> DataVariables { get; }
```
Comment (doddgu) [inline src/Contrib/Configuration/Masa.Contrib.Configuration/Options/MasaAppConfigureOptionsRelation.cs:8]: The pseudocode like:
```csharp
public Dictionary<string, (string ConfigName/SectionName?, string DefaultValue)> Data { get; }
```

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Contrib/Configuration/Masa.Contrib.Configuration/Options/MasaAppConfigureOptionsRelation.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Contrib/Configuration/Masa.Contrib.Configuration/Options/MasaAppConfigureOptionsRelation.cs]: Processed

File: src/Contrib/Configuration/Masa.Contrib.Configuration/Extensions/WebApplicationBuilderExtensions.cs [hunk 3]
```diff
@@ -20,27 +20,27 @@ public static WebApplicationBuilder InitializeAppConfiguration(this WebApplicati
...
                 if (masaConfiguration != null) configuration = masaConfiguration.Local;
-                isInitialize = true;
+                initialized = true;
             }
 
             if (string.IsNullOrWhiteSpace(options.AppId))
-                options.AppId = configuration.GetConfigurationValue(optionsRelation.DataVariables[nameof(options.AppId)],
-                    () => optionsRelation.DataDefaultValue[nameof(options.AppId)]);
+                options.AppId = configuration.GetConfigurationValue(optionsRelation.Data[nameof(options.AppId)].Variable,
+                    () =>  optionsRelation.Data[nameof(options.AppId)].DefaultValue);
```
Comment (doddgu) [inline src/Contrib/Configuration/Masa.Contrib.Configuration/Extensions/WebApplicationBuilderExtensions.cs:35]: An extra space?

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**